src/integration-builder/initialization.js

```diff
@@ -1,29 +1,43 @@
 /* eslint-disable no-undef */
 
-var groupIds = [],
-    consentMapping = {};
-
 var initialization = {
     name: 'OneTrust',
     moduleId: 134,
-    parseConsentGroupIds: function() {
-        groupIds = window.OnetrustActiveGroups ? window.OnetrustActiveGroups.split(',') : [];
-        return groupIds;
+    consentMapping: [],
+    getConsentGroupIds: function() {
+        return window.OnetrustActiveGroups ? window.OnetrustActiveGroups.split(',') : [];
     },
-    parseConsentMapping: function(forwarderSettings) {
-        consentMapping = parseConsentMapping(forwarderSettings.consentGroups) || {};
-        return consentMapping;
+
+    parseConsentMapping: function(rawConsentMapping) {
+        var _consentMapping = {};
+        if (rawConsentMapping) {
+            var parsedMapping = JSON.parse(rawConsentMapping.replace(/&quot;/g, '\"'));
+            parsedMapping.forEach(function(mapping) {
+                _consentMapping[mapping.value] = mapping.map;
+            });
+        }
+
+        return _consentMapping;
     },
+
     initForwarder: function(forwarderSettings) {
         var self = this;
-        this.parseConsentMapping(forwarderSettings);
-        self.parseConsentGroupIds();
-        if (window.Optanon && window.Optanon.OnConsentChanged) {
-            window.Optanon.OnConsentChanged(function() {
-                self.parseConsentGroupIds();
-                self.createConsentEvents();
-            });
-        }
+        self.consentMapping = self.parseConsentMapping(forwarderSettings.consentGroups);
+
+        // Wrap exisitng OptanonWrapper in case customer is using
+        // it for something custom so we can hijack
+        var OptanonWrapperWrapper = window.OptanonWrapper;
```

Comment:
`WrapperWrapper` sounds funny. it's technically a copy of it, so maybe `OptanonWrapperCopy`?

```suggestion
        var OptanonWrapperCopy = window.OptanonWrapper;
```